components/log-viewer-webui/server/src/routes/query.js

```diff
@@ -9,37 +14,52 @@ const EXTRACT_IR_TARGET_UNCOMPRESSED_SIZE = 128 * 1024 * 1024;
  * @return {Promise<void>}
  */
 const routes = async (fastify, options) => {
-    fastify.post("/query/extract-ir", async (req, resp) => {
-        const {origFileId, logEventIdx} = req.body;
+    fastify.post("/query/extract-stream", async (req, resp) => {
+        const {extractJobType, logEventIdx, streamId} = req.body;
         const sanitizedLogEventIdx = Number(logEventIdx);
```

Comment:
I agree, for `extractJobType` and `streamId`.
Let's add this list, to be `export`ed, directly below `QUERY_JOB_TYPE`
```
/**
 * List of valid extract job types.
 */
const EXTRACT_JOB_TYPES = Object.freeze([
    QUERY_JOB_TYPE.EXTRACT_IR,
    QUERY_JOB_TYPE.EXTRACT_JSON,
]);
```
and add a check for `extractJobType` here. e.g.,
```
if (false === EXTRACT_JOB_TYPES.includes(extractJobType)) {
    const err = new Error("Invalid extractJobType");
    err.statusCode = 400;
    throw err;
}
```

We also want to check / sanitize the `streamId` to avoid SQL injection attacks. @kirkrodrigues do you think we should check for the input parameter's length and limit the characters set? Or maybe checking it against a RegExp is efficient.

Reply:
Actually, I was wrong about the SQL injection vulnerability. The `mysqlConnectionPool.query` method should have sanitize the the input to some extend. Let's just check if the `streamId` is a string here.

side question @haiqi96 , does the scheduler hang when the streamId is an invalid string?

Reply:
the scheduler shall return a failed job, saying the streamID doesn't resolve to any archive/file. So it should be fine

Reply:
Sounds like my input is no longer needed, right?

Reply:
Right, we are good now